src/batterystatus_p.h

```diff
@@ -62,13 +62,15 @@ class BatteryStatusPrivate : public QObject
     int chargePercentage;
     int chargeEnableLimit;
     int chargeDisableLimit;
+    bool chargingOverride;
 
 public slots:
     void mceRegistered();
     void mceUnregistered();
 
     void configChanged(const QString &key, const QDBusVariant &value);
     void chargingModeChanged(int mode);
+    void chargingOverrideChanged(const QString override);
```

Comment:
Ampersand missing

Reply:
I think I've been told elsewhere some other time that QString args should be passed by value, not references. But maybe I misunderstood / recall wrong?

Reply:
After some thinking: even if by-value string args were somehow preferable, references work, and internal consistency vs adjacent methods etc wins -> will change.

Reply:
Whenever in doubt, the easiest is to look at some QtCore API - you will notice IN-arguments of a type like QString are passed by const-ref in 99% cases.

Trying hard, some explanation can be found too

- https://wiki.qt.io/ValueBasedAndPointerBasedTypes
- https://doc.qt.io/qt-6/object.html#qt-objects-identity-vs-value

I was sure this was documented a more discoverable way... :\

Reply:
For the record, the size of QString equals the size of a pointer, passing QString by value means incrementing refcount to the actual data. `const` is useful though as it guarantees that the contents doesn't get duplicated as a side effect of an accidental call of a non-const method.

Reply:
In other words the performance impact in most cases is negligible, I often use `QString` for the interface headers to keep them clean and readable, and internally the readability isn't that important, there it could be `const QString` or even `const QString&` to squeeze whatever performance gain can be squeezed out of it. Although performance has become such a non-goal in Sailfish OS that it can't be used as an argument anymore.